apps/nova/src/components/TutorSettings/ProfileSettings.tsx

```diff
@@ -1,24 +1,112 @@
 import { TutorProfileCard } from "@litespace/luna/TutorProfile";
-import React from "react";
-import { useFindTutorInfo } from "@litespace/headless/tutor";
+import React, { useCallback, useMemo } from "react";
 import { Button, ButtonSize } from "@litespace/luna/Button";
 import { useFormatMessage } from "@litespace/luna/hooks/intl";
 
 import { TutorSettingsTabs } from "@/components/TutorSettings/SettingsTabs";
+import { useUpdateUserPersonalInfo } from "@litespace/headless/user";
+import { useToast } from "@litespace/luna/Toast";
+import { useUserContext } from "@litespace/headless/context/user";
+import { Form } from "@litespace/luna/Form";
+import { useForm } from "react-hook-form";
+import { ITutorSettingsForm } from "@/components/TutorSettings/common";
+import { ITutor } from "@litespace/types";
+import { useUserTopics } from "@litespace/headless/topic";
+import { useInvalidateQuery } from "@litespace/headless/query";
+import { QueryKey } from "@litespace/headless/constants";
 
 export const ProfileSettings: React.FC<{
-  id: number;
-}> = ({ id }) => {
-  const tutorInfo = useFindTutorInfo(id);
+  tutorInfo: ITutor.FindTutorInfoApiResponse;
+}> = ({ tutorInfo }) => {
+  const { refetch } = useUserContext();
+  const userTopics = useUserTopics();
   const intl = useFormatMessage();
-  if (!tutorInfo.data) return null;
+  const invalidateQuery = useInvalidateQuery();
+  const toast = useToast();
+
+  const form = useForm<ITutorSettingsForm>({
+    defaultValues: {
+      name: tutorInfo.name || "",
+      bio: tutorInfo.bio || "",
+      about: tutorInfo.about || "",
+    },
+  });
+
+  const name = form.watch("name");
+  const bio = form.watch("bio");
+  const about = form.watch("about");
+
+  const refetchTutorInfo = useCallback(() => {
+    invalidateQuery([QueryKey.FindTutorInfo, tutorInfo.id]);
+    userTopics.refetch();
+    refetch.user();
+  }, [tutorInfo, invalidateQuery, refetch, userTopics]);
```

Comment:
I believe we don't need to call `.refetch` explicitly, you can just call `invalidate` with their query keys without importing their hooks.

Reply:
This should do the same thing as `.refetch()`

```suggestion
  const refetchTutorInfo = useCallback(() => {
    invalidateQuery([QueryKey.FindTutorInfo, QueryKey.FindUserInfo, QueryKey.FindUserTopics]);
  }, [tutorInfo, invalidateQuery, refetch, userTopics]);
```